pkg/actors/actors.go

```diff
@@ -230,6 +235,25 @@ func newActorsWithClock(opts ActorsOpts, clock clock.WithTicker) (ActorRuntime,
 
 	a.timers.SetExecuteTimerFn(a.executeTimer)
 
+	if opts.SchedulerReminders {
+		if opts.Config.SchedulerClients == nil {
+			return nil, fmt.Errorf("scheduler reminders are enabled, but no Scheduler clients are available")
+		}
+		log.Debug("Using Scheduler service for reminders.")
+		// We want to delete "a.actorsReminders" once we move to Scheduler service.
+		a.actorsReminders = reminders.NoOpReminders() // disable old reminder system if using Scheduler for reminders
```

Comment:
Why wasn't this implemented using the `Reminders` interface? Deepanshu and I created that specifically so it could be used by other implementations. We have one and the idea was that the scheduler would use it too (and that's why it can't be deprecated). That would help keeping the code maintainable.

Reply:
Of all funcs in the reminders interface, only 3 relate to scheduler: create, get, delete. It didnt make sense to implement the interface. The abstraction assumes the sidecars are responsible to keeping reminders and triggering them, but distributed scheduler elevates the abstraction.

Reply:
Please still implement the interface. Methods that do nothing can be no-ops. But using the interface allows the code to be much more readable and especially maintainable, since there are different implementations.